torch/csrc/jit/passes/mkldnn_rewrite.cpp

```diff
@@ -6,13 +6,50 @@
 #include <torch/csrc/jit/passes/dead_code_elimination.h>
 #include <torch/csrc/jit/passes/graph_rewrite_helper.h>
 #include <torch/csrc/jit/passes/mkldnn_rewrite.h>
+#include <torch/csrc/jit/passes/mkldnn_rewrite_helper.h>
 #include <torch/csrc/jit/tensorexpr/kernel.h>
 
 namespace torch {
 namespace jit {
 
 #if AT_MKLDNN_ENABLED()
 
+namespace mkldnn {
+
+const std::vector<std::string> zero_scalar_operand =
+    std::vector<std::string>({});
+const std::string algorithm_indicator = std::string("%algorithm");
+
+std::string construct_operand_list(
+    std::vector<std::string> scalar_input,
+    std::string algorithm_indicator) {
+  std::string constructed_operand_list = "";
+
+  std::string joined_scalar_operands = c10::Join(", ", scalar_input);
+  std::string scalar_operands = "%scalars : Scalar?[] = prim::ListConstruct(" +
+      joined_scalar_operands + ")\n";
+
+  constructed_operand_list += scalar_operands;
+
+  if (algorithm_indicator.empty()) {
```

Comment:
If `algorithm_indicator` is non empty, it is not added to the final string. Is it a bug?

Reply:
### `algorithm_indicator` is empty
When there's no `algorithm_indicator` in the input list of the eltwise OP (for example, `relu(Tensor self) -> Tensor`), we need to construct an empty string which will be one of the inputs to the prepack OP. The logic here is to create the string  `%16 : str? = prim::Constant()` in the below IR:
```
%16 : str? = prim::Constant()
%17 : __torch__.torch.classes.mkldnn.ConvOpContext = mkldnn_prepacked::conv2d_prepack(%self.conv.weight, %self.conv.bias, %3, %2, %3, %1, %8, %14, %15, %16)
```

### `algorithm_indicator` is non empty
In the case where `algorithm_indicator` is one of the inputs to the eltwise OP (for example `approximate` in `gelu(Tensor self, *, str approximate='none') -> Tensor`), in the original graph before fusion, we already have this string on the IR since it is input to `gelu`:
```
%1 : str = prim::Constant[value="tanh"]()
%2 = aten::gelu(%14, %1)
```
We don't need to create this string again and will directly use it (`%1` in the below IR) as the last input to the prepack OP:
```
%1 : str = prim::Constant[value="tanh"]()
%17 : __torch__.torch.classes.mkldnn.ConvOpContext = mkldnn_prepacked::conv2d_prepack(%self.conv.weight, %self.conv.bias, %4, %3, %4, %2, %9, %15, %16, %1)
```